### ISSUE TITLE
Named connections don't work with structs

### ISSUE THREAD

**Issue by vovanec** [72squared/redpipe#2]
Hi, 

I'm having trouble when trying to use named connections with structs. The following example with unnamed connection works fine:
```python
import redis
import redpipe

class Counts(redpipe.Struct):

    connection = 'counts'
    keyspace = 'counts'
    key_name = 'name'

    fields = {
        'count': redpipe.IntegerField
    }

def main():
    redpipe.connect_redis(redis.StrictRedis(), name=None)

    with redpipe.pipeline(name=None, autoexec=True) as pipe:
        Counts('1', pipe=pipe, no_op=True).incr('count', 10)

    with redpipe.pipeline(name=None, autoexec=True) as pipe:
        c = Counts('1', pipe=pipe)

    print(c.items())

if __name__ == '__main__':
    main()
```

But it doesn't work when I'm using named connection:

```python
def main():
    redpipe.connect_redis(redis.StrictRedis(), name="counts")

    with redpipe.pipeline(name="counts", autoexec=True) as pipe:
        Counts('1', pipe=pipe, no_op=True).incr('count', 10)

    with redpipe.pipeline(name="counts", autoexec=True) as pipe:
        c = Counts('1', pipe=pipe)
```

Exception is:
```
Vovans-MacBook-Pro:devtools vovan$ python3.4 redpipe_bug.py 
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/redpipe/connections.py", line 51, in get
    return cls.connections[name]()
KeyError: None

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "redpipe_bug.py", line 31, in <module>
    main()
  File "redpipe_bug.py", line 21, in main
    Counts('1', pipe=pipe, no_op=True).incr('count', 10)
  File "/usr/local/lib/python3.4/site-packages/redpipe/structs.py", line 301, in incr
    return new_amount
  File "/usr/local/lib/python3.4/site-packages/redpipe/pipelines.py", line 213, in __exit__
    self.execute()
  File "/usr/local/lib/python3.4/site-packages/redpipe/pipelines.py", line 182, in execute
    promises[0]()
  File "/usr/local/lib/python3.4/site-packages/redpipe/pipelines.py", line 182, in execute
    promises[0]()
  File "/usr/local/lib/python3.4/site-packages/redpipe/pipelines.py", line 150, in process
    pipe = ConnectionManager.get(self.connection_name)
  File "/usr/local/lib/python3.4/site-packages/redpipe/connections.py", line 53, in get
    raise InvalidPipeline('%s is not configured' % name)
redpipe.exceptions.InvalidPipeline: None is not configured
```

Am I doing something wrong?
thanks,
--Vovan

**Comment by 72squared** [comment]
Sorry for the slow reply. I missed the github notification of this issue. Thanks also for the clear repro steps. I'll investigate and see what I find.

**Comment by 72squared** [comment]
Found the bug. Fix coming.

**Comment by 72squared** [comment]
you can see the issue was introduced here:

```
 git diff 931f53549984c63e564f7f679c3b11503ce00331..153a101223c052822e71e7dba3309b5004e60812 -- redpipe/keyspaces.py
```

For some reason I typoed and named the variable `connections` instead of what it should have been: `connection`. Meanwhile, struct used `connection`.

```
 git diff 931f53549984c63e564f7f679c3b11503ce00331..153a101223c052822e71e7dba3309b5004e60812 -- redpipe/structs.py
```

 Since Struct uses hash internally, it wasn't using it's own connection to override the hash object.

Anyway, this issue is resolved in release 1.0.3 which will be coming out shortly.

**Comment by 72squared** [comment]
redpipe release 1.0.3 is now live:

https://pypi.python.org/pypi/redpipe/1.0.3

Let me know if you see any other issues.
